Beta cmdlet docs (#881)
microsoftgraph/entra-powershell: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Set-EntraBetaPolicy.md

# Set-EntraBetaPolicy

## Synopsis

Updates a policy.

## Syntax

```powershell
Set-EntraBetaPolicy 
 -Id <String>
 [-AlternativeIdentifier <String>] 
 [-Definition <System.Collections.Generic.List`1[System.String]>] 
 [-DisplayName <String>] 
 [-Type <String>]
 [-KeyCredentials <System.Collections.Generic.List`1[Microsoft.Open.MSGraph.Model.KeyCredential]>]
 [-IsOrganizationDefault <Boolean>] 
 [<CommonParameters>]
```

## Description

The `Set-EntraBetaPolicy` cmdlet sets a policy in Microsoft Entra ID. Specify `Id` parameter to updates specific policy.

## Examples

### Example 1: Update a policy display name

```powershell

> Connect-Entra -Scopes 'Policy.Read.ApplicationConfiguration, Policy.ReadWrite.ApplicationConfiguration'

Reviewer: ```suggestion
Connect-Entra -Scopes 'Policy.ReadWrite.ApplicationConfiguration'
```

Author: Changes done.

> Set-EntraBetaPolicy -Id 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb' -DisplayName 'NewUpdated'  

Reviewer: ```suggestion
$params = @{
    Id = 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb'
    DisplayName = 'NewUpdated'
}

Set-EntraBetaPolicy @params
```

Author: Changes done.

```

This command updates display name of the specified policy in Microsoft Entra ID.

### Example 2: Update a policy definition

```powershell

> Connect-Entra -Scopes 'Policy.Read.ApplicationConfiguration, Policy.ReadWrite.ApplicationConfiguration'

Reviewer: ```suggestion
Connect-Entra -Scopes 'Policy.ReadWrite.ApplicationConfiguration'
```

Author: Changes done.

> Set-EntraBetaPolicy -Id 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb' -Definition @('{"activityBasedTimeoutPolicies":{"AlternateLoginIDLookup":true, "IncludedUserIds":["UserID"]}}')

Reviewer: ```suggestion
$params = @{
    Id = 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb'
    Definition = @('{"activityBasedTimeoutPolicies":{"AlternateLoginIDLookup":true, "IncludedUserIds":["UserID"]}}')
}

Set-EntraBetaPolicy @params
```

Author: Changes done.

```

This command updates definition of the specified policy in Microsoft Entra ID.

### Example 3: Update a policy organization default

```powershell

> Connect-Entra -Scopes 'Policy.Read.ApplicationConfiguration, Policy.ReadWrite.ApplicationConfiguration'

Reviewer: ```suggestion
Connect-Entra -Scopes 'Policy.ReadWrite.ApplicationConfiguration'
```

Author: Changes done.

> Set-EntraBetaPolicy -Id 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb' -IsOrganizationDefault $false

Author: Changes done.

```

This command updates organization default of the specified policy in Microsoft Entra ID.

### Example 4: Update policy type

```powershell

> Connect-Entra -Scopes 'Policy.Read.ApplicationConfiguration, Policy.ReadWrite.ApplicationConfiguration'

Reviewer: ```suggestion
Connect-Entra -Scopes 'Policy.Read.ApplicationConfiguration, Policy.ReadWrite.ApplicationConfiguration'
```
```suggestion
Connect-Entra -Scopes 'Policy.ReadWrite.ApplicationConfiguration'
```

Author: Changes done.

> Set-EntraBetaPolicy -Id 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb' -Type homeRealmDiscoveryPolicies

Reviewer: ```suggestion
$params = @{
    Id = 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb'
    Type = 'homeRealmDiscoveryPolicies'
}

Set-EntraBetaPolicy @params
```

Author: Changes done.

```

This command updates type of the specified policy in Microsoft Entra ID.

### Example 5: Update policy alternative Identifier

```powershell

> Connect-Entra -Scopes 'Policy.Read.ApplicationConfiguration, Policy.ReadWrite.ApplicationConfiguration'

Reviewer: ```suggestion
Connect-Entra -Scopes 'Policy.ReadWrite.ApplicationConfiguration'
```

Author: Changes done.

> Set-EntraBetaPolicy -Id 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb' -AlternativeIdentifier 'bbbbbbbb-1111-2222-3333-cccccccccccc'

Reviewer: ```suggestion
$params = @{
    Id = 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb'
    AlternativeIdentifier = 'bbbbbbbb-1111-2222-3333-cccccccccccc'
}

Set-EntraBetaPolicy @params
```

Author: Changes done.

```

> This command updates type of the specified policy in Microsoft Entra ID.

Reviewer: ```suggestion
"This example demonstrates how to update the `type` property of a specified policy in Microsoft Entra ID."
```

Author: Changes done.
